vslib/SwitchStateBase.cpp

```diff
@@ -3864,3 +3863,103 @@ sai_status_t SwitchStateBase::queryAttributeCapability(
 
     return SAI_STATUS_SUCCESS;
 }
+
+sai_status_t SwitchStateBase::queryStatsCapability(
+                              _In_ sai_object_id_t switchId,
+                              _In_ sai_object_type_t objectType,
+                              _Inout_ sai_stat_capability_list_t *stats_capability)
+{
+    SWSS_LOG_ENTER();
+    uint32_t i = 0;
+    uint32_t stats_count = 0;
+
+    if (objectType == SAI_OBJECT_TYPE_QUEUE)
+    {
+        stats_count = SAI_QUEUE_STAT_DELAY_WATERMARK_NS;
+        if (stats_capability->count < stats_count )
+        {
+            stats_capability->count = stats_count;
+            return SAI_STATUS_BUFFER_OVERFLOW;
+        }
+
+        stats_capability->count = stats_count;
+
+        for(i = 0; i < stats_capability->count; i++)
+        {
+            stats_capability->list[i].stat_modes = SAI_STATS_MODE_READ_AND_CLEAR | SAI_STATS_MODE_READ ;
+            stats_capability->list[i].stat_enum = i;
+        }
+
+        return SAI_STATUS_SUCCESS;
+    }
+    else if (objectType == SAI_OBJECT_TYPE_PORT)
+    {
+        if (stats_capability->count < 51)
+        {
+            stats_capability->count = 51;
+            return SAI_STATUS_BUFFER_OVERFLOW;
+        }
+
+        stats_capability->count = 51;
```

Comment:
Please use std vector with declared values of that and then assign in loop

Reply:
I have picked only the Ether/If/port in/out and wred stat_enums from the sai_port_stat_t. Hence we need to have unique assignment of these statistics. Please note that i would prefer this way unless it is mandatory to add all the stats in sai_port_stat_t in the capability list.